src/viewport/viewport-matcher-var.directive.ts

```diff
@@ -0,0 +1,87 @@
+import {
+	OnInit,
+	OnDestroy,
+	Directive,
+	Input,
+	ChangeDetectorRef,
+	TemplateRef,
+	ViewContainerRef,
+} from "@angular/core";
+import { combineLatest, ReplaySubject, Subject } from "rxjs";
+import { tap, map, takeUntil } from "rxjs/operators";
+
+import { ViewportService } from "./viewport.service";
+import {
+	isViewportSizeMatcherExpression,
+	isViewportSizeMatcherTupleExpression,
+	isViewportConditionMatch
+} from "./viewport.util";
+import { ViewportMatchConditions } from "./viewport.model";
+
+const NAME_CAMEL = "ssvViewportMatcherVar";
+
+export class SsvViewportMatcherVarContext {
+
+	constructor(
+		public $implicit = false,
+	) { }
+
+}
+
+@Directive({
+	selector: `[${NAME_CAMEL}]`,
+	exportAs: NAME_CAMEL,
+})
+export class SsvViewportMatcherVarDirective implements OnInit, OnDestroy {
+
+	private _matchConditions: ViewportMatchConditions = {};
+	private _context = new SsvViewportMatcherVarContext();
+	private readonly _destroy$ = new Subject<void>();
+	private readonly _update$ = new ReplaySubject<void>(1);
+
+	@Input(`${NAME_CAMEL}When`) set condition(value: string | string[]) {
+		if (isViewportSizeMatcherExpression(value)) {
+			this._matchConditions.expression = value;
+		} else if (isViewportSizeMatcherTupleExpression(value)) {
+			const [op, size] = value;
+			this._matchConditions.expression = {
+				operation: op,
+				size
+			};
+		} else {
+			this._matchConditions.sizeType = value;
+		}
+
+		this._update$.next();
+	}
+
+	constructor(
+		private viewport: ViewportService,
+		private cdr: ChangeDetectorRef,
+		private viewContainer: ViewContainerRef,
+		private templateRef: TemplateRef<SsvViewportMatcherVarContext>,
+	) {
+	}
+
+	ngOnInit(): void {
+		this.updateView();
+		combineLatest([this.viewport.sizeType$, this._update$]).pipe(
+			map(([sizeType]) => isViewportConditionMatch(sizeType, this._matchConditions, this.viewport.sizeTypeMap)),
+			tap(x => this._context.$implicit = x),
+			tap(() => this.cdr.markForCheck()),
```

Comment:
Maybe we can try to call `markForCheck` in the view. I believe you only need a local reference of the view created inside `updateView` and call `markForCheck` on that reference.

So....

```ts
private viewRef: EmbeddedViewRef<SsvViewportMatcherVarContext>;

updateView(): void {
    this.viewContainer.clear();
    this.myViewRef = this.viewContainer.createEmbeddedView(this.templateRef, this._context);
}

ngOnInit(): void {
    ....
    combineLatest([this.viewport.sizeType$, this._update$]).pipe(
			map(([sizeType]) => isViewportConditionMatch(sizeType, this._matchConditions, this.viewport.sizeTypeMap)),
			tap(x => this._context.$implicit = x),
			// tap(() => this.cdr.markForCheck()), // replaced with below + can omit cdr injection
                        tap(() => this.viewRef.markForCheck()),
			takeUntil(this._destroy$),
		).subscribe();
    ....
}
```

Reply:
Yes in fact i have a comment regarding this, was a bit in doubt whats best (sounds better using the viewRef tho)

MarkForCheck tho bubbles upwards, so if you do it in a child component it should bubble up (if I'm correct on this) so it will actually might cause more checks than less; non the less i still think it make sense as well